### DISCUSSION TITLE
Coordinating infrastructure and code changes

### DISCUSSION THREAD

**Discussion by bjartek** [fluxcd/flux2#1178] [Q&A]
Lets say you have an application running that reads an ENV variable in version v1.2.3 and you listen to all the features/bugfixes under the 1 tree with ImageAutomation. 

 Then you introduce a new feature that will require a different value for that ENV value to work. So this is not a breaking change and you can bump the minor version and release v1.3.0. However you do not want the image to be rolled out unless the config value has changed?

One possible way of doing this involves changing the current ImageAutomation first that locks the ceiling to v1.2, then build the image and then in a PR change the ImageAutomation and the value of the config property. Another way might be to pause the ImageAutomation when you build the image. And then when the Image is built you merge the PR in the config repo and then resume the ImageAutomation. 

Have anybody else solved this in a more elegant way? 

Can you configure in the config repo that an ImageAutomation that is manually suspended should be started again?

**Comment by kingdonb** [comment]
What you're describing sounds like a breaking change to me. It's not clear to me if you're saying that v1.3.0 works without the new ENV, it just doesn't enable the new feature? If that's the case, why not roll out the image.

I think you're really asking for fluxcd/flux#224 which raises an alert when an image update can't be pushed past a particular version, or when a "tripwire" is crossed.

**Reply by bjartek** [reply]
I think that issue nails it on the head yes. Basically best practce around how to stop an imageChange until configuration has been updated aswell.
